lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced/search-date-range-advanced.component.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, EventEmitter, Inject, Input, OnDestroy, OnInit, Output, ViewEncapsulation} from '@angular/core';
+import { Subject } from 'rxjs';
+import { endOfDay, format, formatISO, parse, startOfDay } from 'date-fns';
+import { DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE } from '@angular/material/core';
+import { DateFnsAdapter, MAT_DATE_FNS_FORMATS } from '@angular/material-date-fns-adapter';
+import { TranslationService } from '@alfresco/adf-core';
+import { InLastDateType } from './in-last-date-type';
+import { DateRangeType } from './date-range-type';
+import { SearchDateRangeAdvanced } from './search-date-range-advanced';
+import { FormBuilder, Validators } from '@angular/forms';
+import { takeUntil } from 'rxjs/operators';
+import { enUS } from 'date-fns/locale';
+
+const DEFAULT_DATE_DISPLAY_FORMAT: string = 'dd-MMM-yy';
+
+@Component({
+    selector: 'adf-search-date-range-advanced',
+    templateUrl: './search-date-range-advanced.component.html',
+    styleUrls: ['./search-date-range-advanced.component.scss'],
+    providers: [
+        { provide: DateAdapter, useClass: DateFnsAdapter, deps: [ MAT_DATE_LOCALE ] },
+        { provide: MAT_DATE_FORMATS, useValue: MAT_DATE_FNS_FORMATS },
+        { provide: MAT_DATE_LOCALE, useValue: enUS }
```

Comment:
Is that correct? Should be always enUS? What if we are in different localization?

Reply:
As far as I know, the locale is mostly used to drive the date format that gets used for parsing/displaying/etc. Since we are passing (and using) the dateFormat separately as well, I thought it best to just set it as enUS for now. Additionally, the date-fns adapter uses its own object for locale, and not just the locale string, and does not provide any helper methods to get the locale object from the locale string. 

I have however updated the code to set the locale dynamically now, based on what locale the user is currently using/switching to. In order to do this, I also created a helper class called DateFnsUtils with a static method called `getLocaleFromString(locale)`. **This was added to adf-core** so that it is available for others to use as well.